### PR TITLE
use true/false rather than yes/no in the config template

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Changes all `yes` to `true` and `no` to `false` in the config template.

### Motivation

Stay consistent across the config file and avoid confusion with environment variables (`yes` and `no` won't work with env vars).

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -334,22 +334,22 @@ api_key:
...
+# syslog_tls_verify: true
 #
 {{ end -}}
 {{- if .Autodiscovery }}
 # Autodiscovery
 #
 # Change the root directory to look at to get cgroup statistics. Useful when running inside a
 # container with host directories mounted on a different folder.
-# Default if environment variable "DOCKER_DD_AGENT" is set to "yes"
+# Default if environment variable "DOCKER_DD_AGENT" is set to "true"
```
Comment (xvello) [inline pkg/config/config_template.yaml]: For now `yes` is the only valid value for `DOCKER_DD_AGENT`, see https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L565-L568

We could change the check to only check that the envvar is set, like [`IsKubernetes`](https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L565-L568), I'd be all for moving them all to `true` indeed

Comment (arbll) [reply pkg/config/config_template.yaml]: I want to confirm there is zero chance someone changed it to false/no

Comment (xvello) [reply pkg/config/config_template.yaml]: It is an undocumented internal envvar set in the Dockerfile (as it was in Agent5), I think it's fair to assume manually setting it is unsupported.
@hkaj do you think there's a risk here?

**Comment by xvello** [comment]
We also have the `KUBERNETES=yes` and `ECS_FARGATE=yes` envvars for these orchestrators, [the endpoints](https://github.com/DataDog/datadog-agent/tree/master/Dockerfiles/agent/entrypoint) already just check the envvar is set, so changing to `true` in the docs/examples/chart should be seamless. We can take that one.

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (truthbk) [reply pkg/config/config_template.yaml]: I understand the env var doesn't make much sense in a false context, so for more precise language, could we change the comment to:
```
# Default if environment variable "DOCKER_DD_AGENT" is set
```

**Review by truthbk — DISMISSED** [review]
Just a small nit regarding language, for the sake of clarity. Otherwise this looks good to me. 

Can we also reach out to other relevant teams so we can all be in-line with this (if possible) in the config and docs?

**Comment by arbll** [comment]
Reaching out to other team is indeed the next step 🙂 

@xvello All good on your side ?

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (hkaj) [reply pkg/config/config_template.yaml]: It's entirely internal, it's safe to update. We could even update config.go to check for its presence, not its value.

**Review by xvello — DISMISSED** [review]
LGTM !
nit: `docker.IsContainerized` is dead code, superseeded by `config.IsContainerized`, we could just remove it

**Review by xvello — APPROVED** [review]
👍

**Comment by arbll** [comment]
Related docs PR
https://github.com/DataDog/documentation/pull/3000

**Pull request merged**